packages/rum-core/src/rawRumEvent.types.ts

```diff
@@ -209,3 +209,31 @@ export interface CommonContext {
   context: Context
   hasReplay?: true
 }
+
+export interface ReadonlyLocation {
+  readonly hash: string
+  readonly host: string
+  readonly hostname: string
+  readonly href: string
+  readonly origin: string
+  readonly pathname: string
+  readonly port: string
+  readonly protocol: string
+  readonly search: string
+}
+
+export type RumEventDomainContext<E extends RawRumEvent> = E extends RawRumErrorEvent
+  ? { error?: unknown }
+  : E extends RawRumActionEvent
+  ? { event?: Event }
```

Comment:
I didn't add the action name (as initially planned in the RFC) since it is already in the RUM event.